.github/workflows/image.yaml

```diff
@@ -0,0 +1,128 @@
+      - '**'
+
+    tags: # Always, regardless of paths above
+      - '*'
+
+permissions:
+  contents: read
+  packages: write
+  id-token: write
+  security-events: write
+
+jobs:
+  jupyterlab:
+    name: 'Build Docker Images'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        dockerfile:
+          - jupyterlab
+          - jupyterhub
+          - dask-worker
+    steps:
+      - name: 'Checkout Infrastructure'
+        uses: actions/checkout@main
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GitHub Container RegistryT
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GH_TOKEN }}
+
+      - name: Login to Quay Container Registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Lint Dockerfiles
+        uses: jbergstroem/hadolint-gh-action@v1
+        with:
+          dockerfile: Dockerfile.${{ matrix.dockerfile }}
+          output_format: tty
+          error_level: 0
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            "quay.io/nebari/nebari-${{ matrix.dockerfile }}"
+            "ghcr.io/nebari-dev/nebari-${{ matrix.dockerfile }}"
+          tags: |
+            type=ref,event=branch
+            type=ref,event=tag
+            type=sha
+
+      - name: Inspect image dir tree
+        run: |
+          sudo apt-get install tree
+          tree .
+
+      - name: Set tag var
+        id: var
+        run: echo ::set-output name=docker_tag::$(echo "${{ steps.meta.outputs.tags }}" | sed -n 1p)
+
+      - name: Build docker
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: "Dockerfile.${{ matrix.dockerfile }}"
+          tags: ${{ steps.meta.outputs.tags }}
+          push: ${{ github.event_name != 'pull_request' }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - uses: actions/cache@v2.1.4
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          path: .trivy
+          key: ${{ runner.os }}-trivy-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-trivy-
+
+      - name: Run Trivy vulnerability scanner
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ${{ steps.var.outputs.docker_tag }}
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          severity: 'CRITICAL,HIGH'
+          ignore-unfixed: true
+          cache-dir: .trivy
```

Comment:
Now - I get why we are doing the cache step not entirely sure on the value but adding a ping here